Release notes 8.0.19-7 (#1104)
percona/psmdb-docs: docs/release_notes/8.0.19-7.md

title: Percona Server for MongoDB 8.0.19-7 ({{date.8_0_19}})
summary: Learn about improvements, new features, and bug and security vulnerability fixes in this release
authors:
    - Rasika Chivate
version: 8.0.19-7
---

# Percona Server for MongoDB {{ page.meta.version }} ({{date.8_0_19}})

[Install](../install/index.md){.md-button}
[Upgrade from MongoDB Community](../install/upgrade-from-mongodb.md){.md-button}

Percona Server for MongoDB {{ page.meta.version }} is an enhanced, source-available, and highly-scalable database that is a
fully-compatible, drop-in replacement for MongoDB Community Edition.

Percona Server for MongoDB {{ page.meta.version }} includes the improvements and bug fixes of [MongoDB 8.0.18 Community Edition](https://www.mongodb.com/docs/manual/release-notes/8.0/#8.0.18) and [MongoDB 8.0.19 Community Edition](https://www.mongodb.com/docs/manual/release-notes/8.0/#8.0.19)

> Percona Server for MongoDB {{ page.meta.version }} supports protocols and drivers of MongoDB Community 8.0.19.

Reviewer: ```suggestion
Percona Server for MongoDB {{ page.meta.version }} supports protocols and drivers of MongoDB Community 8.0.18 and 8.0.19.
```

- [SERVER-114695](https://jira.mongodb.org/browse/SERVER-114695) 
 **(CVE-2026-1848):** Resolved a vulnerability where connections received on the proxy port (pending proxy protocol headers) were excluded from the `maxConns` limit. This flaw exposed the server to resource exhaustion and OOM (Out-of-Memory) crashes during high-traffic bursts.

- [SERVER-113532](https://jira.mongodb.org/browse/SERVER-113532) 
 **(CVE-2026-1847):** Resolved an issue where inserting specific large documents could prevent secondaries from fetching the oplog from the primary. This could stall replication and potentially lead to node instability.

### Medium severity

- [SERVER-112952](https://jira.mongodb.org/browse/SERVER-112952) 
 **(CVE-2026-25609):** Corrected an issue where improper validation of the profile command caused requests altering the `filter` to be incorrectly treated as read-only.

### Affected versions

These vulnerabilities affect the following versions of MongoDB Community Edition and Percona Server for MongoDB:

> - All Percona Server for MongoDB 8.x versions

Reviewer: ```suggestion
- All Percona Server for MongoDB 8.0.x versions
- MongoDB Community 8.0 versions prior to 8.0.18
```

title: Percona Server for MongoDB 8.0.19-7 ({{date.8_0_19}})
summary: Learn about improvements, new features, and bug and security vulnerability fixes in this release
authors:
    - Rasika Chivate
version: 8.0.19-7
---

# Percona Server for MongoDB {{ page.meta.version }} ({{date.8_0_19}})

[Install](../install/index.md){.md-button}
[Upgrade from MongoDB Community](../install/upgrade-from-mongodb.md){.md-button}

Percona Server for MongoDB {{ page.meta.version }} is an enhanced, source-available, and highly-scalable database that is a
fully-compatible, drop-in replacement for MongoDB Community Edition.

Percona Server for MongoDB {{ page.meta.version }} includes the improvements and bug fixes from:
 
- [MongoDB 8.0.18 Community Edition](https://www.mongodb.com/docs/manual/release-notes/8.0/#8.0.18) 

- [MongoDB 8.0.19 Community Edition](https://www.mongodb.com/docs/manual/release-notes/8.0/#8.0.19)

> Percona Server for MongoDB  {{ page.meta.version }} supports protocols and drivers of MongoDB Community **8.0.18** and **8.0.19**.

Reviewer: please fix this

## New feature

- [PSMDB-1893](https://perconadev.atlassian.net/browse/PSMDB-1893): 
**Disable specific FTDC metric groups**

    Percona Server for MongoDB now permits administrators to disable specific diagnostic metric groups via the new `ftdcMetricGroupsDisabled` parameter. 

       - This parameter can be set in `mongod.conf` 

       - Modified at runtime using `setParameter`

>     By excluding problematic groups, such as `systemMetrics`, `serverStatus.connections`, or `replSetGetStatus` in environments using NFS, FUSE, or autofs, administrators can prevent diagnostic delays caused by unstable or unresponsive mount points. With this approach, the Full Time Diagnostic Data Capture (FTDC) continues to provide essential database insights, such as query performance and replication, without being restrained by external system-level bottlenecks.

Reviewer: ```suggestion
    Database administrators can avoid delays in diagnostics caused by unstable or unresponsive mount points in NFS, FUSE, or autofs environments by excluding certain problematic groups like `systemMetrics`, `serverStatus.connections`, and `replSetGetStatus`. This way, the Full Time Diagnostic Data Capture (FTDC) can still deliver important insights into database performance, such as query efficiency and replication, without being hindered by system-level issues.```

## Security updates: CVE fixes from upstream MongoDB

This release includes upstream MongoDB security fixes for the following vulnerabilities:

### High severity

- [SERVER-116210](https://jira.mongodb.org/browse/SERVER-116210) 

> **(CVE-2026-25611)**: Fixed an issue where a series of specifically crafted, unauthenticated messages could exhaust available system memory, leading to a server crash.

Reviewer: and this

- [SERVER-114838](https://jira.mongodb.org/browse/SERVER-114838) 
 **(CVE-2026-25612):** Resolved a bug in the internal locking mechanism where specific resource encodings could cause unintended collisions between collections. This could lead to resource unavailability due to conflicting locks.

- [SERVER-113685](https://jira.mongodb.org/browse/SERVER-113685) 
 **(CVE-2026-25613):** Fixed a vulnerability where an authorized user could crash the server by querying a collection containing an invalid compound wildcard index.

- [SERVER-102364](https://jira.mongodb.org/browse/SERVER-102364) 
 **(CVE-2026-1849):** Addressed an Out-Of-Memory (OOM) failure during the evaluation of expressions that produce deeply nested documents. The server now properly validates recursion depth to prevent unchecked memory consumption.

- [SERVER-114126](https://jira.mongodb.org/browse/SERVER-114126) 
 **(CVE-2026-1850):** Fixed an issue where complex boolean expression simplifications within the Query Planner by an authorized user could lead to excessive memory usage and an OOM crash.

> - [SERVER-99119](https://jira.mongodb.org/browse/SERVER-99119) **(CVE-2026-25610):** Resolved a vulnerability where an authorized user could trigger a server crash by executing a `$geoNear` aggregation pipeline with specific invalid index hints.

Reviewer: I this this should be fixed, too

> - [SERVER-99119](https://jira.mongodb.org/browse/SERVER-99119) **(CVE-2026-25610):** Resolved a vulnerability where an authorized user could trigger a server crash by executing a `$geoNear` aggregation pipeline with specific invalid index hints.

Author: @radoslawszulgo Please check the rendered output, it's all consistent.

title: Percona Server for MongoDB 8.0.19-7 ({{date.8_0_19}})
summary: Learn about improvements, new features, and bug and security vulnerability fixes in this release
authors:
    - Rasika Chivate
version: 8.0.19-7
---

# Percona Server for MongoDB {{ page.meta.version }} 
({{date.8_0_19}})

[Install](../install/index.md){.md-button}
[Upgrade from MongoDB Community](../install/upgrade-from-mongodb.md){.md-button}

Percona Server for MongoDB {{ page.meta.version }} is an enhanced, source-available, and highly-scalable database that is a
fully-compatible, drop-in replacement for MongoDB Community Edition.

Percona Server for MongoDB {{ page.meta.version }} includes the improvements and bug fixes from:
 

> - [MongoDB 8.0.18 Community Edition](https://www.mongodb.com/docs/manual/release-notes/8.0/#8.0.18) 

Reviewer: ```suggestion
- [MongoDB 8.0.18 Community Edition](https://www.mongodb.com/docs/manual/release-notes/8.0/#8.0.18---jan-27--2026) 
```

> - [MongoDB 8.0.19 Community Edition](https://www.mongodb.com/docs/manual/release-notes/8.0/#8.0.19)

Reviewer: ```suggestion
- [MongoDB 8.0.19 Community Edition](https://www.mongodb.com/docs/manual/release-notes/8.0/#8.0.19---feb-10--2026)
```
